### PR TITLE
128/AddExtensionFeatureToIframesLinksAndApps

### PR DESCRIPTION
**Pull request by alvrba**

### CODE REVIEW

File: src/items/LinkItem.tsx [hunk 1]
```diff
@@ -110,18 +134,20 @@ const LinkItem: FC<LinkItemProps> = ({
       </div>
       <div
         hidden={isLoading}
-        className={classes.iframeContainer}
-        style={{ height: height }}
+        className={clsx(
+          classes.iframeContainer,
+          !isExtendable && classes.iframeContainerWithoutExtension,
```
Comment (pyphilia) [inline src/items/LinkItem.tsx:139]: Here you can use another syntax which allows conditions:
```js
clsx(
    someclass,
    { [classes.xxx]: !isExtendable }
)
```

File: src/items/withExtension.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                  height: height,
+                  x: 0,
+                  y: 0,
+                }}
+                minHeight={IFRAME_MIN_HEIGHT}
+                resizeHandleComponent={{ bottom: <ExtensionIcon /> }}
+                resizeHandleStyles={{ bottom: resizeHandleStyles }}
+              >
+                {component}
+              </Rnd>
```
Comment (pyphilia) [inline src/items/withExtension.tsx:42]: Could we refactor so we use makeStyles?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice PR, actually what about keeping the naming "collapse" instead of extendable? I think they are two opposites to describe the same feature. What do you think?

File: src/icons/ExtensionIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import Box from '@material-ui/core/Box';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import UnfoldMoreIcon from '@material-ui/icons/UnfoldMore';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    icon: {
+      backgroundColor: theme.palette.grey[300],
+      fontSize: '24px',
```
[the remark this answers is not in the document]
Comment (alvrba) [reply src/icons/ExtensionIcon.tsx:10]: Apparently the prop for the size is fontSize (https://v4.mui.com/components/icons/#size)

File: src/items/withExtension.tsx (hunk 2 above)
Comment (alvrba) [reply src/items/withExtension.tsx:42]: I have tried to apply `makeStyles`, but it cannot override the default style. I will ask you tomorrow.

File: src/items/LinkItem.tsx [hunk 4]
```diff
@@ -50,6 +58,7 @@ const LinkItem: FC<LinkItemProps> = ({
   openLinkMessage = 'Click here to open the link manually',
   height: defaultHeight,
   errorMessage = 'The link is malformed.',
+  isExtendable = false,
```
Comment (pyphilia) [inline src/items/LinkItem.tsx:61]: So `isExtendable` is `isResizable`? I think I'm still confused but could we also match the naming with `resizing`? This comment applies to all other files.

File: src/items/LinkItem.tsx [hunk 5]
```diff
@@ -20,13 +22,17 @@ interface LinkItemProps {
...
   errorMessage?: string;
+  isExtendable?: boolean;
 }
 
 const useStyles = makeStyles((theme) => ({
   iframe: {
     width: '100%',
     border: 'none',
+  },
+  iframeWithoutExtension: {
```
Comment (pyphilia) [inline src/items/LinkItem.tsx:33]: `iframeWithoutResizer`?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the changes, I hope we will agree on the naming^^

**Pull request merged**